### PR TITLE
clean up docs on actions, code examples, reference to hybrid

### PR DESCRIPTION
**Pull request by chriselion**
### Proposed change(s)
* Update examples to use the ActionBuffers form of Heuristic and OnActionReceived
* Revise the action documentation. It's still a bit vague - it doesn't actually say "you can use both continuous and discrete", since you can't set that up in the Agent's BrainParameters yet.
* Reworded some of the references to "hybrid".

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: com.unity.ml-agents/Runtime/Actuators/IActionReceiver.cs [hunk 1]
```diff
@@ -102,7 +102,6 @@ internal static ActionBuffers FromActionSpec(ActionSpec actionSpec, float[] acti
                 for (var i = 0; i < actionSpec.NumDiscreteActions; i++)
                 {
                     discreteActions[i] = (int)actions[i + offset];
-
```
Comment (chriselion) [inline com.unity.ml-agents/Runtime/Actuators/IActionReceiver.cs:105]: Some of these changes were auto-formatted. Can revert if they're too noisy.

File: docs/Getting-Started.md [hunk 2]
```diff
@@ -82,11 +82,11 @@ eight elements: the `x` and `z` components of the agent cube's rotation and the
 
 #### Behavior Parameters : Vector Action Space
 
-An Agent is given instructions in the form of a float array of _actions_.
+An Agent is given instructions in the form `ActionBuffers`.
```
Comment (chriselion) [inline docs/Getting-Started.md:85]: `ActionBuffers` might be too advanced for the Getting Started section. Should we just drop this sentence?

File: docs/Learning-Environment-Design-Agents.md [hunk 3]
```diff
@@ -510,20 +510,9 @@ setting the State Size.
 ## Actions
 
 An action is an instruction from the Policy that the agent carries out. The
-action is passed to the Agent as a parameter when the Academy invokes the
-agent's `OnActionReceived()` function. Actions for an agent can take one of two
-forms, either **Continuous** or **Discrete**.
-
-When you specify that the vector action space is **Continuous**, the action
```
Comment (chriselion) [inline docs/Learning-Environment-Design-Agents.md:517]: These sentences are repeated below almost verbatim in their respective sections, so I just removed them.

File: docs/Learning-Environment-Design-Agents.md [hunk 4]
```diff
@@ -553,8 +542,8 @@ up to use either the continuous or the discrete vector action spaces.
 ### Continuous Action Space
 
 When an Agent uses a Policy set to the **Continuous** vector action space, the
```
Comment (chriselion) [inline docs/Learning-Environment-Design-Agents.md:544]: This is the part that I wasn't sure how to rewrite until the BrainParameters refactor is in. Can revisit then.

File: docs/Learning-Environment-Design-Agents.md [hunk 5]
```diff
@@ -619,23 +608,20 @@ gameObject.GetComponent<Rigidbody>().AddForce(
         directionX * 40f, directionY * 300f, directionZ * 40f));
 ```
 
-Note that the above code example is a simplified extract from the AreaAgent
```
Comment (chriselion) [inline docs/Learning-Environment-Design-Agents.md:622]: AreaAgent has been gone for a while.

File: docs/Python-API.md [hunk 6]
```diff
@@ -111,8 +111,8 @@ A `BaseEnv` has the following methods:
   terminates the communication.
 - **Behavior Specs : `env.behavior_specs`** Returns a Mapping of
   `BehaviorName` to `BehaviorSpec` objects (read only).
-  A `BehaviorSpec` contains information such as the observation shapes, the
-  action type (multi-discrete or continuous) and the action shape. Note that
+  A `BehaviorSpec` contains information such as the observation shapes and the
```
Comment (chriselion) [inline docs/Python-API.md:114]: @vincentpierre @andrewcoh Can you review this part and make sure my changes are accurate?

File: docs/Learning-Environment-Design-Agents.md (hunk 4 above)
Comment (andrewcoh) [reply docs/Learning-Environment-Design-Agents.md:544]: maybe we can reword it as "When an Agent's Policy has a **Continuous** vector action space, ..."

File: docs/Learning-Environment-Design-Agents.md [hunk 7]
```diff
@@ -653,26 +639,26 @@ nothing"_ or _"change weapon"_ for his next decision (since action index 1 and 2
...
   branches.
 - You cannot mask all the actions of a branch.
 - You cannot mask actions in continuous control.
 
 ### Actions Summary & Best Practices
 
-- Actions can either use `Discrete` or `Continuous` spaces.
-- When using `Discrete` it is possible to assign multiple action branches, and
-  to mask certain actions.
+- Agents can either use `Discrete` or `Continuous` actions.
```
Comment (andrewcoh) [inline docs/Learning-Environment-Design-Agents.md:654]:
```suggestion
- Agents can use `Discrete` and/or `Continuous` actions.
```

File: docs/Python-API.md (hunk 6 above)
Comment (andrewcoh) [inline docs/Python-API.md:114]:
```suggestion
  A `BehaviorSpec` contains the observation shapes and the
```

Comment (andrewcoh) [reply docs/Python-API.md:114]: Currently the behavior spec just contains those two things, left a suggestion to just explicitly say that.

File: docs/Python-API.md [hunk 8]
```diff
@@ -130,22 +130,18 @@ A `BaseEnv` has the following methods:
...
-  the actions for a whole agent group. `action` is a 2D `np.array` of
-  `dtype=np.int32` in the discrete action case and `dtype=np.float32` in the
-  continuous action case. The first dimension of `action` is the number of
-  agents that requested a decision since the last call to `env.step()`. The
-  second dimension is the number of discrete actions in multi-discrete action
-  type and the number of actions in continuous action type.
+- **Set Actions :`env.set_actions(behavior_name: str, action: ActionTuple)`** Sets
+  the actions for a whole agent group. `action` is an `ActionTuple`, which
+  is made up of a 2D `np.array` of `dtype=np.int32` for discrete action, and
+  `dtype=np.float32` for continuous action case. The first dimension of `np.array`
```
Comment (andrewcoh) [inline docs/Python-API.md:136]:
```suggestion
  is made up of a 2D `np.array` of `dtype=np.int32` for discrete actions, and
  `dtype=np.float32` for continuous actions. The first dimension of `np.array`
```

File: docs/Learning-Environment-Design-Agents.md (hunk 7 above)
Comment (chriselion) [reply docs/Learning-Environment-Design-Agents.md:654]: Might punt on that for now. While technically true, it's only possible with actuators (which we don't have a section on)

**Review by vincentpierre — APPROVED** [review]

File: docs/Python-API.md [hunk 9]
```diff
@@ -166,7 +162,7 @@ A `DecisionSteps` has the following fields :
   for the corresponding Agent. This is used to track Agents across simulation
   steps.
 - `action_mask` is an optional list of two dimensional array of booleans. Only
```
Comment (andrewcoh) [inline docs/Python-API.md:164]:
```suggestion
- `action_mask` is an optional list of two dimensional array of booleans which is only
```

File: docs/Python-API.md [hunk 10]
```diff
@@ -186,7 +182,7 @@ A `DecisionStep` has the following fields:
   the last simulation step.
 - `agent_id` is an int and an unique identifier for the corresponding Agent.
 - `action_mask` is an optional list of one dimensional array of booleans. Only
```
Comment (andrewcoh) [inline docs/Python-API.md:184]:
```suggestion
- `action_mask` is an optional list of one dimensional array of booleans which is only
```

**Review by andrewcoh — APPROVED** [review]
Left some comments/suggestions but LGTM

**Review by dongruoping — APPROVED** [review]

File: docs/Learning-Environment-Design-Agents.md (hunk 4 above)
Comment (chriselion) [reply docs/Learning-Environment-Design-Agents.md:544]: That's better. I think I'll go with "When an Agent's Policy has **continuous** actions..." (trying to get away from "space")

**Pull request merged**